hackernews_tui/src/client/mod.rs

```diff
@@ -376,27 +376,65 @@ impl HNClient {
         let output = std::process::Command::new(&article_parse_command.command)
             .args(&article_parse_command.options)
             .arg(url)
-            .output()?;
-
-        if output.status.success() {
-            match serde_json::from_slice::<Article>(&output.stdout) {
-                Ok(mut article) => {
-                    // Replace a tab character by 4 spaces as it's possible
-                    // that the terminal cannot render the tab character.
-                    article.content = article.content.replace('\t', "    ");
-
-                    article.url = url.to_string();
-                    Ok(article)
-                }
-                Err(err) => {
-                    let stdout = std::str::from_utf8(&output.stdout)?;
-                    warn!("failed to deserialize {} into an `Article` struct:", stdout);
-                    Err(anyhow::anyhow!(err))
+            .output();
+
+        match output {
+            Ok(output) => {
+                if output.status.success() {
+                    match serde_json::from_slice::<Article>(&output.stdout) {
+                        Ok(mut article) => {
+                            // Replace a tab character by 4 spaces as it's possible
+                            // that the terminal cannot render the tab character.
+                            article.content = article.content.replace('\t', "    ");
+
+                            article.url = url.to_string();
+                            Ok(article)
+                        }
+                        Err(err) => {
+                            let stdout = std::str::from_utf8(&output.stdout)?;
+                            warn!("failed to deserialize {} into an `Article` struct:", stdout);
+                            Err(anyhow::anyhow!(err))
+                        }
+                    }
+                } else {
+                    let stderr = std::str::from_utf8(&output.stderr)?.to_string();
+                    Err(anyhow::anyhow!(stderr))
                 }
             }
-        } else {
-            let stderr = std::str::from_utf8(&output.stderr)?.to_string();
-            Err(anyhow::anyhow!(stderr))
+            Err(_) => {
+                // fallback to the `readable-readability` crate if the command fails
+                let html = ureq::AgentBuilder::new()
+                    .timeout(std::time::Duration::from_secs(config::get_config().client_timeout))
+                    .build()
+                    .get(url)
+                    .call()
+                    .map_err(|why| anyhow::anyhow!("failed to get url: {why}"))?
+                    .into_string()
+                    .map_err(|why| anyhow::anyhow!("failed to turn the response into string: {why}"))?;
+                let (nodes, metadata) = readable_readability::Readability::new()
+                    .base_url(url::Url::parse(url)
+                        .map_err(|why| anyhow::anyhow!("failed to parse url: {why}"))?)
```

Comment:
can use `anyhow::Context`, e.g `.context("failed to parse url")` instead.

Reply:
Addressed at [1d03f33](https://github.com/aome510/hackernews-TUI/pull/94/commits/1d03f33a403a36ded9f07de77d7e52186148f376).